hadoop-hdds/common/src/test/java/org/apache/hadoop/ozone/ha/TestFlexibleFQDNResolution.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.ozone.ha;
+
+import org.apache.hadoop.net.NetUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Tests for {@link FlexibleFQDNResolution} class.
+ */
+public class TestFlexibleFQDNResolution {
```

Comment:
Would this work only in k8s environment?

Reply:
By turning on ozone.flexible.fqdn.resolution.enabled, it should work for both traditional servers and k8s based servers. Of course, the traditional servers do not require it. You can turn it off when deploying to the traditional servers